ext/handlers_http.h

```diff
@@ -229,12 +301,27 @@ static inline void ddtrace_inject_distributed_headers_config(zend_array *array,
 
     zval headers;
     ZVAL_ARR(&headers, array);
+    smart_str stream_header = {0};
+    HashTable *header_keys = ddtrace_init_header_keys_set(array, format);
 
 #define ADD_HEADER(header, ...) \
-    if (key_value_pairs) { \
+    if (format == HEADER_MODE_KV_PAIRS) { \
         add_assoc_str_ex(&headers, ZEND_STRL(header), zend_strpprintf(0, __VA_ARGS__)); \
+    } else if (format == HEADER_MODE_ARRAY) { \
+        ddtrace_add_or_update_header_array(Z_ARRVAL(headers), header_keys, ZEND_STRL(header), zend_strpprintf(0, header ": " __VA_ARGS__)); \
     } else { \
-        add_next_index_str(&headers, zend_strpprintf(0, header ": " __VA_ARGS__)); \
+        if (stream_header.s) { \
+            smart_str_appendl(&stream_header, "\r\n", 2); \
+        } \
+        zend_string *formatted = zend_strpprintf(0, header ": " __VA_ARGS__); \
+        smart_str_append(&stream_header, formatted); \
+        \
+        zend_string *key = zend_string_init(ZEND_STRL(header), 0); \
+        zend_str_tolower(ZSTR_VAL(key), ZSTR_LEN(key)); \
+        zend_hash_add_empty_element(header_keys, key); \
+        zend_string_release(key); \
```

Comment:
This is unnecessary, you aren't ever going to insert a same header twice within this function, so we never need to insert these.